[kaixiangg] iP

kaixiangg opened the pull request.

Commit:
Level-0: Rename, Greet, Exit

Commit:
Revert "Level-0: Rename, Greet, Exit"

This reverts commit 4a179d23b9611ffda40273dd2bcbceb79c7c76ee.

Commit:
Add rename, greet and exit features

Commit:
Add echo feature

Commit:
Add ability to "add" whatever text entered by the user and "list" them back to the user when requested

Commit:
Add ability to "mark" and "unmark" tasks as done

Commit:
Add support for tracking "ToDos", "Deadlines" and "Events"

akaash02 commented on src/main/java/Buzz.java:
```diff
@@ -3,6 +3,14 @@
+
 
     //Class-level variables
     private static final Task[] tasks = new Task[100];
```
To follow OOP principles more closely, you can put this in a separate class and create functions to create, modify and delete the elements instead.

akaash02 replied:
Can use ArrayList instead of arrays as they are easier to handle in the future for large use cases.

akaash02 commented on src/main/java/Buzz.java:
```diff
@@ -28,11 +36,43 @@ public static void main(String[] args) {
+                printLine(); // Prints a separator line
+
+                // Splitting the argument using "from" and "to"
+                String[] eventParts = argument.split(" from ", 2);
```
Can put the case for each if else statement in its own separate function to make it easier to edit in the future.

akaash02 commented on src/main/java/Buzz.java:
```diff
@@ -3,6 +3,14 @@
     private static final Scanner scanner = new Scanner(System.in);
+    public static final String NEW_LINE = System.lineSeparator();
+    public static final String SPACE = "      ";
+    public static final String LOGO = " ____        _        \n"
```
Can remove logo if not used.

ljunfun commented on src/main/java/Buzz.java:
```diff
@@ -28,11 +36,43 @@ public static void main(String[] args) {
+
+                // Splitting the argument using "from" and "to"
+                String[] eventParts = argument.split(" from ", 2);
+                if (eventParts.length < 2) {
```
Can include each if-else condition into separate case statements

ljunfun commented on src/main/java/Buzz.java:
```diff
@@ -81,24 +121,29 @@ public static void bye() {
         System.out.println("Buzz: Alright back to Brawl Stars :] byebye!");
```
LGTM!

ljunfun commented on src/main/java/Buzz.java:
```diff
@@ -28,11 +36,43 @@ public static void main(String[] args) {
+                    break;
+                }
+
+                String[] timeParts = eventParts[1].split(" to ", 2);
```
not clear what is timeParts

ljunfun commented on src/main/java/Buzz.java:
```diff
@@ -28,11 +36,43 @@ public static void main(String[] args) {
 
             // Handle the command
             switch (command) {
-            case "add":
```
Good Switch Case Indentation!! :>

ljunfun commented on src/main/java/Buzz.java:
```diff
@@ -81,24 +121,29 @@ public static void bye() {
-            for (int i = 1; i <= taskCount; i++) {
-                System.out.println(i + ".[" + tasks[i - 1].getStatusIcon() + "] " + tasks[i - 1].getDescription());
+            for (int i = 1; i <= taskCount; i += 1) {
+                System.out.println((i) + ". " + tasks[i - 1]);
```
number '1' here is a magic number, can declare as a constant

Commit:
Improved code quality

Major changes include:
Used separate methods for the "Events" and "Deadlines" in the case statements for modularity
Changed to use a magic literal constant of INDEX_OFFSET

Commit:
Improved code quality

Major changes include:
Used separate methods for the "Events" and "Deadlines" in the case statements for modularity
Changed to use a magic literal constant of INDEX_OFFSET

Commit:
Merge branch 'master' of https://github.com/kaixiangg/ip

yeeern27 commented on src/main/java/Buzz.java:
```diff
@@ -77,25 +83,58 @@ public static void bye() {
 
     public static void addTask(String task) {
         if (taskCount < 100) {
```
Great to have follow proper If-else condition coding standard !

yeeern27 commented on src/main/java/Buzz.java:
```diff
@@ -77,25 +83,58 @@ public static void bye() {
-            for (int i = 0; i < taskCount; i++) {
-                System.out.println(i + 1 + ". " + tasks[i]);
+            for (int i = 1; i <= taskCount; i++) {
+                System.out.println(i + ".[" + tasks[i - 1].getStatusIcon() + "] " + tasks[i - 1].getDescription());
```
Line length is too long, try to keep it within 110 - 120 characters.

yeeern27 commented on src/main/java/Buzz.java:
```diff
@@ -5,7 +5,7 @@ public class Buzz {
 
     //Class-level variables
-    private static final String[] tasks = new String[100];
+    private static final Task[] tasks = new Task[100];
```
Plural form was used to name Tasks which represented a list of items, great coding convention followed !

yeeern27 commented on src/main/java/Buzz.java:
```diff
@@ -77,25 +83,58 @@ public static void bye() {
+        }
+    }
+
+    public static void markTaskAsNotDone(String taskNumberStr) {
```
methods named according to what it shall do, easy for understanding, naming convention followed, great !

Commit:
Added error handling features

Commit:
Merge branch 'branch-Level-5'

Commit:
Organise files into packages

Commit:
Merge branch 'branch-A-Packages'

Commit:
Add support for deleting tasks from list

telferang commented on src/main/java/buzz/Buzz.java:
```diff
@@ -0,0 +1,219 @@
+    }
+
+
+    /*Reworked listTasks method to use the toString method in the Task class
+    Just call tasks[i-1] instead of tasks[i-1].getDescription()
+    System.out.println() will implicitly invoke the toString method
+    If task and its subclasses have a toString method, it will be called, and the correctly formatted output will be printed*/
```
Javadoc comment does not follow the coding standards

telferang commented on src/main/java/buzz/Buzz.java:
```diff
@@ -0,0 +1,219 @@
+            throw new BuzzException("Invalid task! Something went wrong while adding it.");
+        }
+
+        if (taskCount >= 100) {
```
Consider having a constant instead to avoid magic numbers

telferang commented on src/main/java/buzz/Buzz.java:
```diff
@@ -0,0 +1,219 @@
+
+
+    //Class-level variables
+    private static final Task[] tasks = new Task[100];
```
Consider having a constant instead to avoid magic numbers

telferang reviewed (commented):
The code is well written. Keep up the good work!

Commit:
Add "save" tasks feature

Commit:
Merge branch 'branch-Level-6'

Commit:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/buzz/Buzz.java

Commit:
Made the code more OOP

Extracted out Ui, Parser, Storage, TaskList for more OOP

Commit:
Merge branch 'branch-moreOOP'

Commit:
Added feature to find tasks

Find tasks using keywords typed in by the user

Commit:
Add JavaDoc comments

kaixiangg committed:
Merge pull request #1 from kaixiangg/branch-Level-9

Added feature to find tasks

Commit:
Merge branch 'master' into branch-A-JavaDoc

kaixiangg committed:
Merge pull request #2 from kaixiangg/branch-A-JavaDoc

Add JavaDoc comments

Commit:
Updated README user guide

Refactored the User Guide from Duke to Buzz, and included the relevant features of Buzz ChatBot
